source/opt/ir_context.h

```diff
@@ -53,11 +54,34 @@ class IRContext {
   friend inline Analysis operator<<(Analysis a, int shift);
   friend inline Analysis& operator<<=(Analysis& a, int shift);
 
+  // Trivial constructor for testing purposes
```

Comment:
What goes wrong if someone uses this by accident?  Seems like we should more strenuously steer people away from this.

If you truly need something for testing purposes, then I recommend making a static method to make it, eg..

static IRContext* MakeIRContextForTest() ... and then hide the details


...
Then I read the tests and made a suggestion there to use a helper function to make the dummy IRContexts.